components/connector/Dockerfile

```diff
@@ -6,9 +6,9 @@ WORKDIR ${BASE_APP_DIR}
 COPY ./internal/ ${BASE_APP_DIR}/internal/
 COPY ./config/ ${BASE_APP_DIR}/config/
 COPY ./pkg/ ${BASE_APP_DIR}/pkg/
-COPY ./vendor/ ${BASE_APP_DIR}/vendor/
 COPY ./cmd/main.go ${BASE_APP_DIR}/
 COPY ./licenses ${BASE_APP_DIR}/licenses
+COPY go.mod go.sum ${BASE_APP_DIR}/
```

Comment:
Why is this necessary? I don't think I find it in our old go mod components?

Reply:
Most of the other go modules components copy the whole current directory (including go.mod and go.sum) in Docker image while in connector's Dockerfile only some of the directories were copied (e.g. /hack is not copied). They need to be in the image so that go knows which are the correct dependencies of this mod project.

Reply:
Since we're unifying the approach and if all other components copy the whole directory, but not this one - why not unify it here as well (it seems like a good opportunity while you're at it)

At the end of the day it will end up making a separate Docker stage which will copy only the binary, so I don't see much overhead in unifying this Dockerfile like most others since any additional files that you may add here will be scrapped after the final stage is created.

Reply:
Can't we just add everything `COPY . ${BASE_APP_DIR}` and if there's something that needs to be excluded (e.g. `/hack`), we can add it in a `.dockerignore` file. Do you guys like the idea? It'll look a little bit cleaner.

Reply:
Yes, that's exactly what all our other components do `COPY . ${BASE_APP_DIR}` - and that was what @krasish was referencing and what would be needed if we unify the approaches.
Also, like I mentioned in the previous comment - the end game is to build some binaries which you can copy to the second Docker stage which is a clean slate, so it doesn't really matter if you add a /hack directory or not it will ultimately get discarded by the second stage as it only contains the binary file.
Of course if there's some logically reason that you want the first stage to ignore some files, you can always use `.dockerignore` but I'm not aware of any potential issues which may mandate something like that at this point.

Reply:
I agree it will give a unified look at our components and it looks cleaner. I also cannot find a specific reason to copy folders one by one instead of the whole current directory. I will change that.

Reply:
There is a reason to copy only the dependencies first as it will give the ability to not rebuild all dependencies when only the code is changed. I will leave it like this and even change it in other places to first copy the go.mod and go.sum

Reply:
Hmm... aren't the dependencies going to be pulled(rebuild) always even when only the code is changed in the `RUN go build -v -o main .` command as the Docker container will not have the dependencies (starting from scratch always)?

Reply:
if we have `RUN go mod vendor/download` before `go build`, then go build will not try to download dependencies.

Reply:
Makes sense now:)

Reply:
I now understand what @pankrator meant. I agree that this will be a significant improvement in our image. Thanks for the comments!